dlib/dnn/layers.h

```diff
@@ -1522,6 +1564,37 @@ namespace dlib
 
             unsigned long new_window_size;
         };
+
+        class visitor_bn_input_no_bias
+        {
+        public:
+
+            template <typename T>
+            void set_input_no_bias(T&) const
+            {
+                // ignore other layer types
+            }
+
+            template <layer_mode mode, typename U, typename E>
+            void set_input_no_bias(add_layer<bn_<mode>, U, E>& l)
+            {
+                disable_bias(l.subnet().layer_details());
+                set_bias_learning_rate_multiplier(l.subnet().layer_details(), 0);
+                set_bias_weight_decay_multiplier(l.subnet().layer_details(), 0);
```

Comment:
I know these two functions are redundant, but I thought it was good to keep them any way.

Reply:
Yeah, seems reasonable.